### PR TITLE
Fix loading datasets `traffic_2008_10T` and `traffic_2008_hourly`

### LINKED ISSUE #455: [BUG] Loading of some internal datasets fails
### 🐛 Bug Report

Can't load following internal datasets:
```
traffic_2008_10T
traffic_2008_hourly
```

### Expected behavior

Datasets are loading correctly

### Code
```
from etna.datasets import load_dataset

load_dataset("traffic_2008_hourly", rebuild_dataset=True)
```

### Environment

_No response_

### Additional context

```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
Cell In[85], line 3
      1 from etna.datasets import load_dataset
----> 3 load_dataset("traffic_2008_hourly", rebuild_dataset=True)

File ~/etna/etna/datasets/internal_datasets.py:159, in load_dataset(name, download_path, rebuild_dataset, parts)
    156 freq = dataset_params["freq"]
    158 if not _check_dataset_local(dataset_path) or rebuild_dataset:
--> 159     get_dataset_function(dataset_dir)
    160 ts_out = []
    161 for part in parts_:

File ~/etna/etna/datasets/internal_datasets.py:363, in get_traffic_2008_dataset(dataset_dir, dataset_freq)
    360 dates_df = dates_df.drop(["dt"], axis=1)
    362 df = pd.DataFrame(targets, columns=stations)
--> 363 df["timestamp"] = dates_df_cropped["timestamp"].values
    364 df = df.merge(dates_df, on=["timestamp"], how="right").fillna(0)
    365 df = df.melt("timestamp", var_name="segment", value_name="target")

File ~/Library/Caches/pypoetry/virtualenvs/etna-yJQui467-py3.8/lib/python3.8/site-packages/pandas/core/frame.py:3980, in DataFrame.__setitem__(self, key, value)
   3977     self._setitem_array([key], value)
   3978 else:
   3979     # set column
-> 3980     self._set_item(key, value)

File ~/Library/Caches/pypoetry/virtualenvs/etna-yJQui467-py3.8/lib/python3.8/site-packages/pandas/core/frame.py:4174, in DataFrame._set_item(self, key, value)
   4164 def _set_item(self, key, value) -> None:
   4165     """
   4166     Add series to DataFrame in specified column.
   4167 
   (...)
   4172     ensure homogeneity.
   4173     """
-> 4174     value = self._sanitize_column(value)
   4176     if (
   4177         key in self.columns
   4178         and value.ndim == 1
   4179         and not is_extension_array_dtype(value)
   4180     ):
   4181         # broadcast across multiple columns if necessary
   4182         if not self.columns.is_unique or isinstance(self.columns, MultiIndex):

File ~/Library/Caches/pypoetry/virtualenvs/etna-yJQui467-py3.8/lib/python3.8/site-packages/pandas/core/frame.py:4915, in DataFrame._sanitize_column(self, value)
   4912     return _reindex_for_setitem(Series(value), self.index)
   4914 if is_list_like(value):
-> 4915     com.require_length_match(value, self.index)
   4916 return sanitize_array(value, self.index, copy=True, allow_2d=True)

File ~/Library/Caches/pypoetry/virtualenvs/etna-yJQui467-py3.8/lib/python3.8/site-packages/pandas/core/common.py:571, in require_length_match(data, index)
    567 """
    568 Check the length of data matches the length of the index.
    569 """
    570 if len(data) != len(index):
--> 571     raise ValueError(
    572         "Length of values "
    573         f"({len(data)}) "
    574         "does not match length of index "
    575         f"({len(index)})"
    576     )

ValueError: Length of values (63648) does not match length of index (63360)
```

### Checklist

- [X] Bug appears at the latest library version

### PR DESCRIPTION
**Pull request by d-a-bunin**
## Before submitting (must do checklist)

- [ ] Did you read the [contribution guide](https://github.com/etna-team/etna/blob/master/CONTRIBUTING.md)?
- [ ] Did you update the docs? We use Numpy format for all the methods and classes.
- [ ] Did you write any new necessary tests?
- [x] Did you update the [CHANGELOG](https://github.com/etna-team/etna/blob/master/CHANGELOG.md)?

## Proposed Changes
Fix holidays dates. It seems like on new version of `holidays` the holidays aren't correct. There were also incorrect holidays with missing values. The new values are taken from the [source](https://archive.ics.uci.edu/dataset/204/pems+sf)

## Closing issues
Closes #455.

### CODE REVIEW

File: etna/datasets/internal_datasets.py [hunk 1]
```diff
@@ -348,11 +347,21 @@ def read_data(path: Path, part: str) -> np.ndarray:
     targets = np.concatenate([targets_train, targets_test], axis=0)
     targets = targets[np.argsort(ts_indecies)].reshape(-1, 963)
 
-    drop_days = (
-        list(holidays.country_holidays(country="US", years=2008).keys())
-        + list(holidays.country_holidays(country="US", years=2009).keys())[:3]
-        + [date(2009, 3, 8), date(2009, 3, 10)]
-    )
+    drop_days = [
```
Comment (brsnw250) [inline etna/datasets/internal_datasets.py:350]: Can we add comments describing what happening here?

File: etna/datasets/internal_datasets.py [hunk 2]
```diff
@@ -348,11 +347,21 @@ def read_data(path: Path, part: str) -> np.ndarray:
...
+        date(2008, 7, 4),
+        date(2008, 9, 1),
+        date(2008, 10, 13),
+        date(2008, 11, 11),
+        date(2008, 11, 27),
+        date(2008, 12, 25),
+        date(2009, 1, 1),
+        date(2009, 1, 19),
+        date(2009, 2, 16),
+    ] + [date(2008, 3, 8), date(2009, 3, 9)]
```
Comment (brsnw250) [inline etna/datasets/internal_datasets.py:364]: We can include this dates in the list above, maybe add a comment if needed

File: tests/test_datasets/test_internal_datasets.py [hunk 3]
```diff
@@ -81,181 +81,189 @@ def test_not_present_part():
...
-            marks=pytest.mark.skip(reason="Dataset is too large for testing in GitHub."),
-        ),
-        (
-            "m4_quarterly",
-            (866 + 8, 24000),
-            0,
-            873,
-            ("train", "test"),
-        ),
+        # pytest.param(
```
Comment (brsnw250) [inline tests/test_datasets/test_internal_datasets.py:84]: Why all this test cases are commented ?

**Review by brsnw250 — CHANGES_REQUESTED** [review]

File: tests/test_datasets/test_internal_datasets.py (hunk 3 above)
Comment (d-a-bunin) [reply tests/test_datasets/test_internal_datasets.py:84]: It is my mistake, I'll fix it.

**Review by brsnw250 — APPROVED** [review]

**Comment by egoriyaa** [comment]
May be we can write issue to `holidays` github about wrong dates?

**Comment by d-a-bunin** [comment]
Created a bug: https://github.com/vacanza/python-holidays/issues/1963

**Pull request merged**